ssh2/sftp_handle.pyx

```diff
@@ -270,7 +270,16 @@ cdef class SFTPHandle:
         cdef char *cbuf = buf
         cdef ssize_t rc
         with nogil:
-            rc = c_sftp.libssh2_sftp_write(self._handle, cbuf, _size)
+            while _size >= 0:
+                if _size == 0:
```

Comment:
Might be better to just set `cdef ssize_t rc = 0` by default and let it fall through when `size > 0` fails rather than have to do conditional checks. What do you think?